File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const { length } = nums;
  const map = new Map();

  for (let i = 0; i < length; i++) {
    if (map.has(nums[i])) return [map.get(nums[i]), i];
    map.set(target - nums[i], i);
  }

  return [-1, -1];
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int n = nums[i];
            if (map.contains(n)) return {map[n], i};
            map[target - n] = i;
        }

        return {-1, -1};
    }
};